### DISCUSSION TITLE
Shipping tax calculation

### DISCUSSION THREAD

**Discussion by sstottelaar** [duncanmcclean/simple-commerce#1065] [Help] [answered]
Hi there,

I want to implement shipping tax in my store, but I'm having trouble with the setup.

In my config I have this enabled:

```
'tax_engine_config' => [
        // Basic Engine
        'rate' => 21,
        'included_in_prices' => false,
        'shipping_taxes' => true,

        // Standard Tax Engine
        'address' => 'billing',
```

This is my tax setup in Statamic:

<img width="912" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/11455085/4342f559-7939-4edc-ae3d-86bc428878a4">

The applied rate should be 21%, but in the calculation, I get a different percentage than expected  (approx 17,39%):

<img width="347" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/11455085/bd0dcff1-50b8-4e1d-be36-351ff9d55f43">

My shipping rate is €2,99 (excl. tax) and the calculated total with tax is €3,51.

Am I missing something? Any help is appreciated, thanks.

**Comment by duncanmcclean** [comment]
If you create a product with the same price, do you get the same calculation? What's the "inc tax" total you're expecting?

**Reply by sstottelaar** [reply]
Sorry for the delay, but the issue remains. This time on another site. Say I want to ensure the total shipping cost (including 21% tax) to be exactly €10, how would I set this up? I've tried multiple things:

- Set the `calculateCost` in my shipping method to `1000`
- Set the `calculateCost` in my shipping method to `826` (€10 - 21% tax)
- Toggled between `Prices include tax?` in my default shipping tax rate on/off

Take a look at this example:

<img width="419" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/11455085/c131c058-9e38-4630-9e48-c372298ffe47">

In this example the cost of my shipping method is set to `826` with these settings:

<img width="922" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/11455085/2bf2a4fb-39ef-4ad2-9baf-f48643c764bf">

I'd expect the `shipping_total_with_tax` to be exactly €10, but somehow it calculates €9,69.

Any idea what I'm doing wrong here?

EDIT:

When setting `Prices include tax?` to true I get the following:

<img width="420" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/11455085/b7ccf842-68da-471d-a212-e101cf22e3da">

The total sum of the cart is as expected now (€20 for the cart + €10 shipping), but the values for `shipping_total` and `shipping_total_with_tax` are equal. My expectation would be that `shipping_total_with_tax` would be €10 in this case.

**Reply by duncanmcclean** [reply, marked as answer]
I've just tried to replicate this scenario in my test site and it seems to be working fine for me....

<img width="260" alt="image" src="https://github.com/duncanmcclean/simple-commerce/assets/19637309/bbcd07f6-9375-4e45-8eca-d9a8fe8d1c49">

_^ using `{{ sc:cart:shipping_total }}`, `{{ sc:cart:shipping_total_with_tax }}` and `{{ sc:cart:tax_total }}` in this screenshot._

The shipping tax rate is set to 21% and my shipping method is returning `1000`. 🤔 

Could it be using one of the other tax rates you have configured instead? If you change the other tax rates to 0% and change something on the order to let it recalculate (like resubmitting the "select shipping method" form), does tax still get applied to shipping or not?

> but the values for shipping_total and shipping_total_with_tax are equal. My expectation would be that shipping_total_with_tax would be €10 in this case.

Whoops, that's a bug. I've fixed it in #1095, which will be included in a release shortly.

_As an FYI: I'm hoping to re-work the tax system in a future version, the current system can be pretty confusing, even for me and I built the thing._ 😂

**Reply by sstottelaar** [reply]
Thank you for looking into it, `shipping_total_with_tax` now returns the expected value.

And I'm all for making it less confusing haha, but it seems to work fine. Cheers!
